storage/imxrt-flash/fspi.c

```diff
@@ -109,41 +76,104 @@ static int flexspi_checkFlags(flexspi_t *fspi)
 }
 
 
-static ssize_t flexspi_opRead(flexspi_t *fspi, time_t start, struct xferOp *xfer)
+static int flexspi_poll(flexspi_t *fspi, volatile uint32_t *addr, uint32_t mask, uint32_t val, time_t start, time_t timeout, int checkFlags)
 {
 	int res;
-	uint8_t *ptr = xfer->data.read.ptr;
-	uint8_t *end = ptr + xfer->data.read.sz;
+	int retries = 0;
 
-	while (ptr != end) {
-		volatile uint8_t *rfdr = (volatile uint8_t *)(fspi->base + rfdr32); /* 2x */
-
-		/* Wait for rx FIFO available */
-		while ((*(fspi->base + intr) & (1u << 5u)) == 0u) {
+	while ((*addr & mask) != val) {
+		if (checkFlags) {
```

Comment:
Use `bool checkFlags`, or:
```suggestion
		if (checkFlags != 0) {
```